MSYS2 / MinGW, GCC 15.1: conflicting types in getopt

```
[20/30] Compiling C object bin/fribidi.exe.p/getopt.c.obj
FAILED: bin/fribidi.exe.p/getopt.c.obj 
"gcc.bat" "-Ibin/fribidi.exe.p" "-Ibin" "-I../bin" "-I." "-I.." "-Ilib" "-I../lib" "-Igen.tab" "-I../gen.tab" "-fdiagnostics-color=always" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-O3" "-D_FORTIFY_SOURCE=2" "-fstack-protector-strong" "-mtune=generic" "-O2" "-pipe" "-D__USE_MINGW_ANSI_STDIO=1" "-mthreads" "-DHAVE_CONFIG_H" "-DFRIBIDI_LIB_STATIC" -MD -MQ bin/fribidi.exe.p/getopt.c.obj -MF "bin/fribidi.exe.p/getopt.c.obj.d" -o bin/fribidi.exe.p/getopt.c.obj "-c" ../bin/getopt.c
../bin/getopt.c: In function '_getopt_initialize':
../bin/getopt.c:291:28: error: too many arguments to function 'getenv'; expected 0, have 1
  291 |   d->__posixly_correct = !!getenv ("POSIXLY_CORRECT");
      |                            ^~~~~~  ~~~~~~~~~~~~~~~~~
../bin/getopt.c:146:14: note: declared here
  146 | extern char *getenv (
      |              ^~~~~~
../bin/getopt.c: At top level:
../bin/getopt.c:1185:1: error: conflicting types for 'getopt'; have 'int(int,  char * const*, const char *)'
 1185 | getopt (
      | ^~~~~~
In file included from ../bin/getopt.c:98:
../bin/getopt.h:158:14: note: previous declaration of 'getopt' with type 'int(void)'
  158 |   extern int getopt (
      |              ^~~~~~
```

[logs.zip](https://github.com/user-attachments/files/19931199/logs.zip)

--- a/bin/getopt.c
+++ b/bin/getopt.c
@@ -50,13 +50,13 @@
 
 #ifndef ELIDE_CODE
 
+#include <stdlib.h>
 
 /* This needs to come after some library #include
    to get __GNU_LIBRARY__ defined.  */
 #ifdef	__GNU_LIBRARY__
-/* Don't include stdlib.h for non-GNU C libraries because some of them
+/* Don't include unistd.h for non-GNU C libraries because some of them
    contain conflicting prototypes for getopt.  */
-# include <stdlib.h>
 # include <unistd.h>
 #endif /* GNU C library.  */
 
@@ -137,18 +137,6 @@ int optopt = '?';
 static struct _getopt_data getopt_data;
 
 
-#ifndef __GNU_LIBRARY__
-
-/* Avoid depending on library functions or files
-   whose names are inconsistent.  */
-
-#ifndef getenv
-extern char *getenv (
-);
-#endif
-
-#endif /* not __GNU_LIBRARY__ */
-
 #ifdef _LIBC
 /* Stored original parameters.
    XXX This is no good solution.  We should rather copy the args so

--- a/bin/getopt.h
+++ b/bin/getopt.h
@@ -145,20 +145,6 @@ extern "C"
    arguments to the option '\0'.  This behavior is specific to the GNU
    `getopt'.  */
 
-#ifdef __GNU_LIBRARY__
-/* Many other libraries have conflicting prototypes for getopt, with
-   differences in the consts, in stdlib.h.  To avoid compilation
-   errors, only prototype getopt for the GNU C library.  */
-  extern int getopt (
-  int ___argc,
-  char *const *___argv,
-  const char *__shortopts
-  ) __THROW;
-#else				/* not __GNU_LIBRARY__ */
-  extern int getopt (
-  );
-#endif				/* __GNU_LIBRARY__ */
-
 #ifndef __need_getopt
   extern int getopt_long (
   int ___argc,

